### PR TITLE
Allowing for custom imageio namespace and other dev amenities

### LINKED ISSUE #319: Support a custom namespace of kubevirt/CDI deployments for testing
There are several steps required ATM to make a functional tests work with a namespace other than 'cdi' for CDI.
When deploying KubeVirt using HCO, all of the components including CDI will be created under the kubevirt-hyperconverged.
For that reason, the certificates created on https://github.com/machacekondra/imageiotest and https://github.com/machacekondra/fakeovirt will fail to work since they assume the server/CN to be imageio.cdi.

CDI project addressed this issue by creating the certificates as part of the init-container so they can be accessed and copied to fakeovirt and imageiotest containers: 
https://github.com/kubevirt/containerized-data-importer/blob/master/manifests/templates/imageio.yaml.in#L40

Additional support will be required on the test-side for the following resources:

- https://github.com/kubevirt/vm-import-operator/blob/master/cluster/manifests/imageio.yaml will have to support a custom namespace and be extended to include the init-container for creating the certificates
- Both URL and ca.pem should be changed according to the generated CA cert and the namespace for cdi/kubevirt: https://github.com/kubevirt/vm-import-operator/blob/master/tests/cirros/secret.yml#L9
- https://github.com/kubevirt/vm-import-operator/blob/master/tests/ovirt/ovirt-consts.go#L5 should be adjusted to support other namespace and ca.pem

### PR DESCRIPTION
**Pull request by jakub-dzon**
Fixes #319 
```release-note
NONE
```

Signed-off-by: Jakub Dzon <jdzon@redhat.com>

### CODE REVIEW

**Comment by masayag** [comment]
Please describe the steps required to run functional tests again external cluster https://github.com/kubevirt/vm-import-operator/blob/master/docs/developers.md#functional-testing
Also, was there any change needed for imagetest or fakeovirt ?

**Comment by jakub-dzon** [comment]
There were no changes required in fakeovirt or imageiotest. 
I will update the documentation.

**Comment by jakub-dzon** [comment]
@masayag - I've updated the documentation; please check if it works with your setup

**Review by masayag — APPROVED** [review]

File: automation/execute-tests.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+echo "  DEFAULT_SC: $DEFAULT_SC"
+echo "  NFS_SC: $NFS_SC"
+
+./cluster/imageio-install.sh "$IMAGEIO_NAMESPACE"
+
+FAKEOVIRT_CA_PATH=${FAKEOVIRT_CA_PATH:-$(pwd)/_out/fakeovirt-ca.pem}
+
+IMAGEIO_POD_NAME=$($KUBECTL get pods --no-headers -o custom-columns=":metadata.name" --selector='app=imageio' -n "$IMAGEIO_NAMESPACE")
+echo "Imageio pod: $IMAGEIO_POD_NAME"
+$KUBECTL -n "$IMAGEIO_NAMESPACE" exec "$IMAGEIO_POD_NAME" -c imageiotest -- cat /tmp/certs/ca.pem > "$FAKEOVIRT_CA_PATH"
```
Comment (machacekondra) [inline automation/execute-tests.sh:35]: What about using here: `... exec deploy/imageio-deployment -c imageiotest`, instead of pod name?

**Review by machacekondra — APPROVED** [review]

File: automation/execute-tests.sh (hunk 1 above)
Comment (jakub-dzon) [reply automation/execute-tests.sh:35]: Thanks! That would be much simpler!

**Pull request merged**